### ISSUE TITLE
fit$summary() throws error

### ISSUE THREAD

**Issue by bob-carpenter** [stan-dev/cmdstanr#383]
**Describe the bug**

```
> fit <- mod$sample(data = data,
                  init = 0.25,
                  chains = 1, parallel_chains = 4,
                  iter_warmup = 100, iter_sampling = 100,
		  refresh = 1)	     

...reporting on chains running...

> > fit$summary()
Error in UseMethod("variables") : 
  no applicable method for 'variables' applied to an object of class "NULL"
```

**To Reproduce**

See above.

**Expected behavior**

Print fit object.

**Operating system**

```
> sessionInfo()
R version 4.0.1 (2020-06-06)
Platform: x86_64-apple-darwin17.0 (64-bit)
Running under: macOS Catalina 10.15.7

Matrix products: default
BLAS:   /System/Library/Frameworks/Accelerate.framework/Versions/A/Frameworks/vecLib.framework/Versions/A/libBLAS.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/4.0/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] cmdstanr_0.2.1

loaded via a namespace (and not attached):
 [1] ps_1.3.3          crayon_1.3.4      R6_2.4.1          lifecycle_0.2.0   jsonlite_1.6.1   
 [6] backports_1.1.8   posterior_0.1.0   magrittr_1.5      pillar_1.4.4      rlang_0.4.6      
[11] rstudioapi_0.11   data.table_1.13.0 vctrs_0.3.1       checkmate_2.0.0   ellipsis_0.3.1   
[16] tools_4.0.1       yaml_2.2.1        abind_1.4-5       compiler_4.0.1    processx_3.4.2   
[21] pkgconfig_2.0.3   tibble_3.0.1     
```

**CmdStanR version number**

See above.

**Additional context**

Downloaded cmdstan using the built-in function from CmdStanR and rebuilt to deal with PCH issue.

**Comment by rok-cesnovar** [comment]
Thanks for the report. 

Are you still in this state? Can you check fit$output_files() to print the resulting CSV files and see if the CSV files look correct?

**Comment by jgabry** [comment]
@rok-cesnovar I think this error is because we need to require a newer version of the posterior package. I don't know what changed in cmdstanr that requires a newer version of posterior, but we always test against the latest posterior so that's probably why this wasn't caught by our tests. We may need to start testing against multiple versions of posterior to avoid something like this. 

@bob-carpenter Can you try updating the posterior package with  `devtools::install_github("stan-dev/posterior")` and then see if the error goes away? Sorry for the hassle.

**Comment by mitzimorris** [comment]
Bob has gone back to using RStan

**Comment by rok-cesnovar** [comment]
>  I think this error is because we need to require a newer version of the posterior package.

But we didn't change anything in posterior apart from some speedup? Or did I miss anything?

**Comment by jgabry** [comment]
We must have changed something in cmdstanr? I tried installing the same version of posterior bob had and I got that error.

**Comment by rok-cesnovar** [comment]
Ah yes, https://github.com/stan-dev/cmdstanr/blob/master/DESCRIPTION#L30 we still point to 0.1.0

**Comment by rok-cesnovar** [comment]
> We may need to start testing against multiple versions of posterior to avoid something like this.

Will add that.

**Comment by bob-carpenter** [comment]
> Are you still in this state? Can you check fit$output_files() to print the resulting CSV files and see if the CSV files look correct?

I closed, but can try to recreate if you still need me to.

> Bob has gone back to using RStan

That's because I have a project I need to get done and couldn't get CmdStanR installed and working properly.  I filed the issue because I hope to use it when I can get it installed.

**Comment by jgabry** [comment]
> That's because I have a project I need to get done and couldn't get CmdStanR installed and working properly. I filed the issue because I hope to use it when I can get it installed.

Sorry I completely misunderstood! Thanks for offering to recreate but I think we figured it out. It should be fixed by bumping the required version of posterior in https://github.com/stan-dev/cmdstanr/commit/9eda8f959886b1392f45eb1f81fe35e9df878740. If you want to try it out either of these commands should install the proper versions of everything now: 

```r
install.packages("cmdstanr", repos = c("https://mc-stan.org/r-packages/", getOption("repos")))
```

or 

```r
# install.packages("remotes")
remotes::install_github("stan-dev/cmdstanr")
```

**Comment by bob-carpenter** [comment]
> @bob-carpenter Can you try updating the posterior package with devtools::install_github("stan-dev/posterior") and then see if the error goes away? Sorry for the hassle.

```
> devtools::install_github("stan-dev/posterior")
Error in loadNamespace(name) : there is no package called ‘devtools’
```

Did I need to have devtools installed to install CmdStanR?  I tried `install.packages('devtools')` which errored out with:

```
Error: cannot remove prior installation of package ‘processx’
In addition: Warning message:
In doTryCatch(return(expr), name, parentenv, handler) :
  unable to load shared object '/Library/Frameworks/R.framework/Resources/modules//R_X11.so':
  dlopen(/Library/Frameworks/R.framework/Resources/modules//R_X11.so, 6): Library not loaded: /opt/X11/lib/libSM.6.dylib
  Referenced from: /Library/Frameworks/R.framework/Versions/4.0/Resources/modules/R_X11.so
  Reason: image not found

> library('devtools')
Error in library("devtools") : there is no package called ‘devtools’
```

This is indicative of the kind of problems I have trying to follow the diagnostics.  I'm not an R-dev, just a novice user.

**Comment by jgabry** [comment]
Ugh yeah those diagnostics are pretty useless! 

> Did I need to have devtools installed to install CmdStanR?

No, that's just one option. You can also use 

```r
install.packages("cmdstanr", repos = c("https://mc-stan.org/r-packages/", getOption("repos")))
```

without devtools. Maybe try this: 

```r
# terminate and restart R to make sure no packages are loaded
remove.packages(c("posterior", "cmdstanr", "processx"))

```r
install.packages("cmdstanr", repos = c("https://mc-stan.org/r-packages/", getOption("repos")))
````

**Comment by jgabry** [comment]
> > Bob has gone back to using RStan
> 
> That's because I have a project I need to get done and couldn't get CmdStanR installed and working properly. I filed the issue because I hope to use it when I can get it installed.

Sorry I thought we had lost you ;) I know you've run into a bunch of issues that still need to be sorted out with CmdStanR.

**Comment by jgabry** [comment]
I'm going to close this because the issue with summary() is fixed by 9eda8f9, but @bob-carpenter I can definitely keep trying to help troubleshoot the install trouble from https://github.com/stan-dev/cmdstanr/issues/383#issuecomment-738290585 if it's still not working. I'm sorry it's been such a hassle!
